### PR TITLE
Feature - PHP client overhaul

### PR DESCRIPTION
**Pull request by ATimmeh33**

### CODE REVIEW

File: signhost.php [hunk 1]
```diff
@@ -345,11 +345,11 @@ function __construct($type) {
 	}
 }
 
-class iDEAL extends Verification implements JsonSerializable {
+class IDealVerification extends Verification implements JsonSerializable {
```
Comment (WouterJanson) [inline signhost.php]: Doesn't changing the classname cause backwards compatibility issues?
Or do we accept this as long it isn't a package on composer yet?

File: signhost.php [hunk 2]
```diff
@@ -230,73 +222,49 @@ class Signer implements JsonSerializable {
 
 	function __construct(
```
Comment (WouterJanson) [inline signhost.php]: Again, I think we accept the compatibility break for now?

**Review by WouterJanson — CHANGES_REQUESTED** [review]
In general, the changes look good, as far as I'm concerned. But there are some things that would break backwards compatibility, do we want to allow this for now? Until it's available via Composer?


Can't comment on this part, as it isn't touched but you refactored all other comments:

```php
// Returns binary stream
return $response;
```

Probably better to also use phpDoc on the methods, indicating the return type.
```php
 /**
  * @return type
  */
```

And maybe also the params, if it's being touched anyway.

File: signhost.php (hunk 2 above)
Comment (ATimmeh33) [reply signhost.php]: Yes, since we are not using SemVer breaking changes are okay.

File: signhost.php [hunk 3]
```diff
@@ -216,21 +309,64 @@ function jsonSerialize() {
...
+	public $Reference;
+
+	/** @var string */
+	public $ReturnUrl;
 
+	/** @var object */
+	public $Context;
+
+	/**
+	 * @param string         $email,
```
Comment (WouterHD) [inline signhost.php]: Comment ends in a ,

**Review by WouterHD — CHANGES_REQUESTED** [review]
Looking good (apart from that single , in your comment)

File: signhost.php [hunk 4]
```diff
@@ -1,10 +1,48 @@
...
+require_once("DTO/scribbleverification.php");
+require_once("DTO/signer.php");
+require_once("DTO/signingcertificateverification.php");
+require_once("DTO/surfnetverification.php");
+require_once("DTO/transaction.php");
+require_once("DTO/verification.php");
+
+class SignhostApiClient {
+	const API_VERSION = "v1";
+	const USER_AGENT_IDENTIFIER = "Signhost PHP Client v0.1";
```
Comment (MrJoe) [inline signhost.php]: Does not conform to the HTTP standard.

**Comment by WouterHD** [comment]
:+1:

**Review by WouterHD — APPROVED** [review]
:+1:

**Review by WouterHD — APPROVED** [review]

**Pull request merged**